update aptos.toml

hhuang728 opened the pull request:
add new repo without depulicated repo in origianl toml file

hhuang728 commented:
Hey @jubos , could you please help to review about PR? 

Also I have a question about the sub-ecosystem developer/commit calculation? For example, pancakeswap is the sub ecosystem for aptos. But when EC count commits or developers for aptos, did EC count all commits from pancakeswap repos? Since pancakeswap is the sub-ecosystem for many chains, I am a little bit confuse about the calculation logic. Thanks!

hhuang728 commented:
Hey @jubos  , could you please help to review this PR? Thank you!

jubos reviewed (changes requested):
@hhuang728, thank you for the PR.

As of now, this casts a bit too wide of a net.  There are a lot of developer forks of aptos-core with no changes, so we tend not to add those because they can pollute the taxonomy.

Can you scope this change and add some of the easier ones (e.g. repos that have aptos in the name or have many mentions of Aptos) and then do a separate PR for some of the sub ecosystem additions like Wormhole or the Arweave related repos that may or may not be Aptos related?

hhuang728 commented:
Hi @jubos , thanks for your reply!
just want to clarify more detail here:
1. would you like us to remove forked repo from aptos-core?
2. for _'add some of the easier ones (e.g. repos that have aptos in the name or have many mentions of Aptos)'_, 'have many mentions of Aptos', could you please let me know how to find 'mentions' here? 
3. a separate PR for some of the sub ecosystem.
(1)would you like to put repos which belong to sub-ecosystem to sub ecosystem toml, such as wormhole-foundation/relayer-network-dashboard will be put in wormhole.toml?
(2)how to deal with the repo name which including sub ecosystem name, such as ciphermarco/wormhole or gabzim/wormhole-relayer?

jubos commented:
> Hi @jubos , thanks for your reply! just want to clarify more detail here:
> 
>     1. would you like us to remove forked repo from aptos-core?
Yes, unless they have significant commits that are different from aptos-core. 

>     2. for _'add some of the easier ones (e.g. repos that have aptos in the name or have many mentions of Aptos)'_, 'have many mentions of Aptos', could you please let me know how to find 'mentions' here?

You can do a grep of aptos in the codebase or in the README.md.  Some of the repos that might be pulling in a transitive dependency on a project that has aptos mentioned somewhere would have a very tenuous connection to Aptos and probably shouldn't be included.

>     3. a separate PR for some of the sub ecosystem.
>        (1)would you like to put repos which belong to sub-ecosystem to sub ecosystem toml, such as wormhole-foundation/relayer-network-dashboard will be put in wormhole.toml?
>        (2)how to deal with the repo name which including sub ecosystem name, such as ciphermarco/wormhole or gabzim/wormhole-relayer?

Those should be added into Wormhole and Wormhole can be a sub ecosystem of Aptos instead of adding them directly to the aptos.toml file.

hhuang728 commented:
Hi @jubos, Thanks for providing detail feedback. I just updated the repo list based on above discussion. However there is Error: Process completed with exit code 2 in the validation check. Could you please help to take a look? Thanks!

hhuang728 commented:
Hi @jubos ,got it. just submitted based on all previous feedback. Please help to review. Thank you!

hhuang728 commented:
Hey @jubos , thanks for all of your previous help! 
could you please help review this updated pr based on previous guideline?

jubos commented:
Hi @hhuang728, thanks for all the fixes here.  It's just a really tough one to review with so many additions.  Just from an anecdotal sampling, there still appears to many non aptos repos in here.  Some examples are:

https://github.com/Webners1/Scroll-Swap
https://github.com/xChatGTP/server
https://github.com/InjectiveLabs/injective-ts
https://github.com/web3helpers/wallets-hooks
https://github.com/xIceyyx/CantoFaucet

I would suggest paring it down and limit it to just repos that have "aptos" in their name and/or their README directly as a starting point.  Then we can get that merged and you can come back with a deeper search on some of these other ones that are on the borderline.  Reviewing 7000 additions is tricky without some automation work to ensure that they are indeed aptos repos.

jubos reviewed (changes requested):
Please see my comment on the PR

hhuang728 commented:
Hi @jubos , thank you so much for providing above suggestion!

I have updated the commit with repos that include "aptos" in their name or their README file. I added 729 repo here. Please help to review. Thanks.

jubos commented on data/ecosystems/a/aptos.toml:
```diff
@@ -608,280 +770,236 @@ url = "https://github.com/aptos-foundation/node-tools"
 
 [[repo]]
 url = "https://github.com/aptos-labs/actions-old"
-tags = ["Protocol"]
```
Can you leave these Protocol tags in?

jubos reviewed (changes requested):
Soo close!  Thanks for these changes.  Can you leave the missing and tag attributes that were removed? 

Then I think it is ready to go.

hhuang728 commented:
Hi @jubos , just added back missing and tags attributes. Please help to review, thanks again!

jubos commented:
Thanks @hhuang728 for making all of the necessary changes!

hhuang728 commented:
Hi @jubos, thanks for your supporting for all above changes!
for previous borderline repos, I scanned the codebase and README.md for each repo and picked repos with more than 5 times mention in either codebase or README.md. 
eg: 
1.https://github.com/Webners1/Scroll-Swap, there is an ’aptos’ folder(https://github.com/Webners1/Scroll-Swap/tree/master/apps/aptos) and mentioned many aptos’ in the repo.
2.https://github.com/xChatGTP/server, there are some aptos mentioned https://github.com/xChatGTP/server/blob/main/yarn.lock
3.https://github.com/web3helpers/wallets-hooks, there is a aptos wallet folder https://github.com/web3helpers/wallets-hooks/tree/main/packages/aptos-wallet

Could you please help to figure out how to decide borderline repos here? Thank you!
